### PR TITLE
(test) Use kebab-case naming convention and BDD-like syntax

### PR DESCRIPTION
**Pull request by denniskigen**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR refactors e2e tests in this repo to use the kebab-case [naming convention](https://o3-docs.vercel.app/docs/coding-conventions#naming). Also tweaks a couple of expectations to use stricter assertions.

## Related issue 

https://issues.openmrs.org/browse/O3-2172

### CODE REVIEW

File: e2e/specs/edit-form.spec.ts [hunk 1]
```diff
@@ -5,27 +5,28 @@ import {
...
   await addFormResources(api, valueReference, form.uuid);
 });
 
-test("Should be able to edit a form", async ({ page }) => {
+test("should edit a form", async ({ page }) => {
   const formBuilderPage = new FormBuilderPage(page);
-
   await formBuilderPage.gotoFormBuilder();
-  // Click on the edit schema button
+  await expect(page.getByTestId(`editSchema${form.uuid}`)).toBeVisible();
```
Comment (anjula-sack) [inline e2e/specs/edit-form.spec.ts:23]: @denniskigen Ideally what needs to happen is to check all the table pages for the relevant form. Currently it just checks the first page

File: e2e/specs/create-form-using-custom-schema.spec.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    await formBuilderPage.schemaInput().fill(JSON.stringify(customSchema));
+    // Inputs the custom schema and render changes
+    await formBuilderPage.renderChangesButton().click();
+  });
+
+  await test.step("Then I click the `Save Form` button", async () => {
+    await formBuilderPage.saveForm();
+  });
+
+  await test.step(" I should get a success message and be redirected to the edit page for the new form", async () => {
```
Comment (jayasanka-sack) [inline e2e/specs/create-form-using-custom-schema.spec.ts:30]:
```suggestion
  await test.step("And I should get a success message and be redirected to the edit page for the new form", async () => {
```

steps should start with either one of the following term:

- Given
- When
- Then
- And
- But

I'll update the guide

Comment (jayasanka-sack) [reply e2e/specs/create-form-using-custom-schema.spec.ts:30]: ex: https://docs.behat.org/en/v2.5/guides/1.gherkin.html

File: e2e/specs/create-form-using-custom-schema.spec.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+let formUuid = null;
+
+test("Create a form using a custom JSON schema", async ({ page }) => {
+  const formBuilderPage = new FormBuilderPage(page);
+
+  await test.step("When I visit the form builder", async () => {
+    await formBuilderPage.gotoFormBuilder();
+  });
+
+  await test.step("And click the `Create New Form` button", async () => {
```
Comment (jayasanka-sack) [inline e2e/specs/create-form-using-custom-schema.spec.ts:16]:
```suggestion
  await test.step("And I click the `Create New Form` button", async () => {
```

**Review by anjula-sack — COMMENTED** [review]
@denniskigen can you link this issue https://issues.openmrs.org/browse/O3-2172 this was created earlier.

**Comment by denniskigen** [comment]
@anjula-sack Done.

**Review by jayasanka-sack — APPROVED** [review]
Thanks, @denniskigen !

**Pull request merged**